holoviews/core/data/xarray.py

```diff
@@ -93,7 +93,7 @@ def retrieve_unit_and_label(dim):
         if isinstance(data, xr.DataArray):
             kdim_len = len(kdim_param.default) if kdims is None else len(kdims)
             vdim_len = len(vdim_param.default) if vdims is None else len(vdims)
-            if vdim_len > 1 and kdim_len == len(data.dims)-1 and data.shape[-1] == vdim_len:
+            if kdim_len == len(data.dims)-1 and data.shape[-1] == vdim_len:
```

Comment:
Why is `vdim_len` not needed anymore?

Reply:
This was the proximate cause of the problem in the first place, as I described above categorical arrays with a single category were treated differently from categorical arrays with multiple dimensions. We now handle them consistently.